lightning/src/ln/channel.rs

```diff
@@ -1701,7 +1750,8 @@ impl<ChanSigner: ChannelKeys> Channel<ChanSigner> {
 				removed_outbound_total_msat += htlc.amount_msat;
 			}
 		}
-		if htlc_inbound_value_msat + msg.amount_msat + self.value_to_self_msat > (self.channel_value_satoshis - Channel::<ChanSigner>::get_our_channel_reserve_satoshis(self.channel_value_satoshis)) * 1000 + removed_outbound_total_msat {
+		let remote_fee_cost = self.commit_tx_fee_inbound_htlc();
+		if htlc_inbound_value_msat + msg.amount_msat + self.value_to_self_msat > (self.channel_value_satoshis - Channel::<ChanSigner>::get_our_channel_reserve_satoshis(self.channel_value_satoshis)) * 1000 + removed_outbound_total_msat - remote_fee_cost {
```

Comment:
A few notes:
 * we need to be absurdly careful with overflows here - to make it more Obvious (tm) that this cannot overflow, probably best to move the fee cost to the other side and add it instead of subtracting,
 * I think this would be more readable if the commit_tx_fee_inbound_htlc returned the HTLC count, instead of it also doing math internally, which I think would have made the above off-by-one more obvious.

Reply:
Fyi, I personally rewrote as our_personal_value - pending_to_their < channel_value - reserve - fees, as "Our channel balance must be not be superior at available channel value"

Reply:
FWIW, just seeing this comment, but my take on it is https://github.com/rust-bitcoin/rust-lightning/pull/577#discussion_r418281825, mostly for readability. :)

Would be fine with moving `remote_fee_cost_msat` over to the other side to be more obvious that the overflow won't occur. May read better that way, too, now that I think of it.